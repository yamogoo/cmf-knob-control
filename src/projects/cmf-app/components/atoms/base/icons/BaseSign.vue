<template lang="pug">
div(:class="[CLASS_NAME, `${CLASS_NAME}--${size}`]")
    BaseIcon(
        :name="name"
    )
</template>

<script setup lang="ts">
import { APP_PREFIX } from '@app/config';
import BaseIcon, { Icons } from './BaseIcon.vue';

const CLASS_NAME = `${APP_PREFIX}-sign`;

interface Props {
    name: Icons;
    size?: Sizes;
}

withDefaults(defineProps<Props>(), {
    size: Sizes.SM
});
</script>

<script lang="ts">
export { Icons } from './BaseIcon.vue';

export enum Sizes {
    SM = 'sm',
    MD = 'md',
    LG = 'lg'
}
</script>

<style lang="scss">
.#{$APP_PREFIX}-sign {

    &--sm {
        @include box(32px);
    }

    &--md {
        @include box(36px);
    }

    &--lg {
        @include box(40px);
    }
}
</style>