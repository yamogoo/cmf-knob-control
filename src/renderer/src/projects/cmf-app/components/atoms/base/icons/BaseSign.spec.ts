import { mount } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'

import BaseSign, { Icons } from './BaseIcon.vue'

describe('BaseSign', () => {
  test.each([Icons.ASSISTED, Icons.ATTENTION])('should render Icon (name="%s")', async (name) => {
    const component = mount(BaseSign, {
      props: {
        name
      }
    })

    await vi.dynamicImportSettled()

    const iconEl = component.find('svg')
    const isIconElExist = iconEl.exists()

    expect(isIconElExist).toBe(true)
    expect(isIconElExist).toMatchSnapshot()
  })
})
