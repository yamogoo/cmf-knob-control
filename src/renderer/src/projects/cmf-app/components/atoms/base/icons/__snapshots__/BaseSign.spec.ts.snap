// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseSign > should render Icon (name="AttentionSymbol") 1`] = `true`;

exports[`BaseSign > should render Icon (name="ProgramsSymbol") 1`] = `true`;
